pennylane_lightning/src/bindings/Bindings.cpp

```diff
@@ -324,6 +324,29 @@ void lightning_class_bindings(py::module &m) {
                 const std::vector<size_t> &wires) {
                  return M.expval(operation, wires);
              })
+        .def("generate_samples",
+             [](Measures<PrecisionT> &M, size_t num_shots) {
+                 auto &&result = M.generate_samples(num_shots);
+                 // int num_samples = num_shots;
+                 // std::vector<int> result(num_samples*wires.size(),0);
+
+                 int num_wires = result.size() / num_shots;
+                 ssize_t ndim = 2;
+                 std::vector<ssize_t> shape = {static_cast<ssize_t>(num_shots),
+					       static_cast<ssize_t>(num_wires)};
+                 std::vector<ssize_t> strides = {static_cast<ssize_t>(sizeof(int) * num_wires),
+						 static_cast<ssize_t>(sizeof(int))};
+
+                 // return 2-D NumPy array
+                 return py::array(py::buffer_info(
+                     result.data(), /* data as contiguous array  */
+                     sizeof(int),   /* size of one scalar        */
+                     py::format_descriptor<int>::format(), /* data type */
```

Comment:
```suggestion
                     py::format_descriptor<size_t>::format(), /* data type */
```

Again, feel free to ignore if `int` are a hard requirement, but if not better to use `size_t`.